azero/contracts/ownable/lib.rs

```diff
@@ -0,0 +1,129 @@
+#![cfg_attr(not(feature = "std"), no_std, no_main)]
+
+/// This crate facilitates introducing the ownership of contracts and ownership changes using a two-step process.
+///
+/// The basic functionality is similar to the `Ownable` concept and exposes the following messages/methods:
+/// * `get_owner`
+/// * `is_owner`
+/// * `ensure_owner`: a helper to use with the `?` syntax that will check whether the caller is the owner of the contract
+///
+/// Additionally, it introduces the following method for transferring ownership:
+/// * `transfer_ownership`: callable only by the current owner, appoints the new owner but instead of making them the owner right away, it stores them in the `pending_owner` field
+/// * `accept_owership`: callable only by the pending owner, removes the previous owner and makes them the sole owner of the contract
+/// * `get_pending_owner`: returns the pending owner, if the ownership change process is currently underway.  
+///
+/// In order to use it in your contract, implement the methods of the `Ownable2Step` trait: in most cases, you can simply call the corresponding methods on the `Data` object.
+use ink::primitives::AccountId;
+use scale::{Decode, Encode};
+
+#[derive(Debug, PartialEq, Eq, Encode, Decode)]
+#[cfg_attr(feature = "std", derive(scale_info::TypeInfo))]
+pub enum Error {
+    /// The caller didn't have the permissions to call a given method
+    UnauthorizedAccount(AccountId),
```

Comment:
Would change to `CallerNotOwner`, but just my preference, can be kept the way it is